### ISSUE TITLE
Offline and Element overlay capability question

### ISSUE THREAD

**Issue by vessp** [Telerik-Verified-Plugins/Mapbox#34]
Hello there, i'm starting a new project for which I need to use maps on both ios and android and this plugin looks amazing.  However I need to show images overtop of the map sometimes, is this possible?.  I've seen some talk about putting the webview on top of the map view but not sure if that has been done yet. 

Also, MapBox native libraries [have support for offline usage](https://www.mapbox.com/help/mobile-offline/) and I was wondering if this is supported yet.

Thanks for a great plugin!

**Comment by EddyVerbruggen** [comment]
So are you going to use [NativeScript](https://github.com/EddyVerbruggen/nativescript-mapbox/issues/13) or Cordova? 😄

**Comment by vessp** [comment]
Well at this point it could be either.  I haven't used NativeScript before, so using Cordova (maybe with ionic) would be easier for me.

**Comment by zenzora** [comment]
I'm wondering the same thing, trying to use cordova

**Comment by anothar** [comment]
Well, I use Cordova with Ionic for my application. Works very well except earlier versions of android: 4.0-4.1. If you need to show images as markers - you can use [my fork](https://github.com/anothar/Mapbox/tree/features_image_marker) . I hope it will be merged into master. Regarding general overlay - you can trace [progress and discussion](https://github.com/Telerik-Verified-Plugins/Mapbox/pull/29).

**Comment by zenzora** [comment]
Ok thanks, how about offline tiles?

**Comment by anothar** [comment]
Here is [discussion](https://github.com/Telerik-Verified-Plugins/Mapbox/issues/19) and [pull request](https://github.com/Telerik-Verified-Plugins/Mapbox/pull/22). So both features are work in progress.

**Comment by zenzora** [comment]
Awesome, thanks!

**Comment by EddyVerbruggen** [comment]
@anothar I will surely merge your awesome PR! Quite a coincidence that I'm currently working on custom marker support in the NativeScript plugin. You implementation using SVGs is a bit more sophisticated though ;)

The only problem I see is those other PR's you mentioned. Things are really getting in each others way.. I think I'll first merge one or two of the other and then add yours as it seems easier to integrate and I don't want to stress those other forks by merging this in atm.

**Comment by anothar** [comment]
> You implementation using SVGs is a bit more sophisticated though ;)

Yeah, I wish I knew simpler way. Compiling library in Xcode made me crazy ;)

>  I don't want to stress those other forks by merging this in atm.

Ok!

**Comment by anothar** [comment]
@EddyVerbruggen maybe it'll be better to create for those PR's new branch-version 2.0 or just develop because they break current API? And then reimplement custom markers in that branch (it'll be simple). So that new features can be implemented and merged in that branch without fear of breaking plugin work.

**Comment by gls-labs** [comment]
Hello, I would like to know if is possible to implement offline maps using cordova/ionic with Telerik Mapbox Plugin. Thanks!
